timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.java

```diff
@@ -517,11 +518,13 @@ private static void checkExceptionLogging(JavaContext context, UCallExpression c
         }
       }
 
-      String s = evaluateString(context, arg2, true);
-      if (s == null || s.isEmpty()) {
-        LintFix fix = quickFixIssueExceptionLogging(arg2);
-        context.report(ISSUE_EXCEPTION_LOGGING, arg2, context.getLocation(call),
-            "Use single-argument log method instead of null/empty message", fix);
+      if (!(arg2 instanceof UReferenceExpression)) {
```

Comment:
`ConstantEvaluator.evaluateString` used below only evaluates constants. If the argument is not constant it is always null, thus causing false positives. 

Having this check prevents the issue but it checks less now. 

So before it was able to catch below usage.

```java
String someMessage = null;
Timber.d(exception, someMessage);
```

This will not be reported anymore. Only simple usages like `Timber.d(exception, null);` will be reported. And that is what is covered by the tests anyways.

Reply:
Great start, but the regression can be avoided by checking whether the `UReferenceExpression` (specifically, a `USimpleNameReferenceExpression`) resolved to a Field type, i.e., "does this reference refer to a field?"  See #280.